### DISCUSSION TITLE
Browsershot::html not working with Remote Chrome

### DISCUSSION THREAD

**Discussion by LittleToni** [spatie/browsershot#486] [Q&A]
When I try to combine ```Browsershot::html``` with ```Browsershot::setRemoteInstance``` I run into the following error.

```
Spatie\Browsershot\Exceptions\CouldNotTakeBrowsershot
For some reason Chrome did not write a file at `<file_name>`.
```

Example:

```
Browsershot::html('<h1>test<h1>')
    ->setRemoteInstance('chrome', 3000)
    ->noSandbox()
    ->save('<file_name>');
```

I think the problem is that puppeteer's page.goto('file://<file_path>') not work because the file is not available on the remote Chrome. Would it be better to read the temporary html file and use puppeteer's `page.setContent`-method?

**Comment by hendrikheil** [comment]
We just ran into the same issue.
It's indeed related to the `goto` call. However when we changed `browser.js` to use `setContent` instead, we had to make it wait for `networkIdle0` (to load all external resources like fonts) which is not ideal, because it adds at least 500ms of latency to the request.

So instead we just proxied the rendered views through a route on the same service so we can just provide a URL.
We saw average request times of around 1000ms with `url` compared to 2400ms with `setContent`.

**Comment by pablodiegoss** [comment]
On #491 I've shared a little bit of my experience with browsershot remote instance and docker. Looks like you are trying to access chrome remotely with a hostname, that only works with [an IP](https://spatie.be/docs/browsershot/v4/miscellaneous-options/connection-to-a-remote-chrome-instance).

https://github.com/spatie/browsershot/discussions/491#discussioncomment-8383609

**Comment by webrcost4** [comment]
Hi! This is a common issue when combining `Browsershot::html()` with a remote Chrome instance (`setRemoteInstance`). Here’s what’s happening:

### 🔹 The problem
- `Browsershot::html('<h1>test</h1>')` generates a **temporary local HTML file**.  
- `setRemoteInstance('chrome', 3000)` runs Chrome on a **different host** (remote machine or container).  
- Chrome **cannot access local files** via `file://` on a different host, so it fails to save the output and throws:


This is why your current approach with `file://` does not work on a remote Chrome.

---

### 🔹 The correct approach

Instead of relying on a temporary file, you should **send the HTML content directly to Chrome** using Puppeteer’s `page.setContent()` (Browsershot supports this via `setHtml()`):

```php
use Spatie\Browsershot\Browsershot;

Browsershot::remoteChrome('http://remote-chrome:3000')
    ->setHtml('<h1>test</h1>')  // pass HTML directly
    ->noSandbox()
    ->save('test.pdf');          // or test.png
